parallel-cluster-post-install.sh

```diff
@@ -68,87 +88,271 @@ sed -i "s|@CUDA_INSTALL_PATH@|${CUDA_INSTALL_PATH}|g" /etc/profile.d/cuda.sh
 PATH=$PATH:@CRYOSPARC_INSTALL_PATH@/cryosparc_master/bin
 EOF
 sed -i "s|@CRYOSPARC_INSTALL_PATH@|${CRYOSPARC_INSTALL_PATH}|g" /etc/profile.d/cryosparc.sh
 . /etc/profile.d/cryosparc.sh
 
-echo "export CRYOSPARC_FORCE_HOSTNAME=true" >> "${CRYOSPARC_INSTALL_PATH}"/cryosparc_master/config.sh
-echo "export CRYOSPARC_DISABLE_IMPORT_ON_MASTER=true" >> "${CRYOSPARC_INSTALL_PATH}"/cryosparc_master/config.sh
+# Condition checks whether /etc/profile.d/cryosparc.sh activated previously install cryosparc
+# if not, then we install cryosparc
+if [ ! -x "$(command -v "cryosparcm")" ]; then
+  echo "Installing fresh CryoSPARC"
+
+  # Download cryoSPARC
+  mkdir -p "${CRYOSPARC_INSTALL_PATH}"
+  # Need to make sure OSUSER can write to this path
+  chown ${OSUSER}:${OSGROUP} ${CRYOSPARC_INSTALL_PATH}
+
+  cd "${CRYOSPARC_INSTALL_PATH}" || return
+  [ -f "${CRYOSPARC_INSTALL_PATH}/cryosparc_master.tar.gz" ] || curl -L "https://get.cryosparc.com/download/master-v4.0.3/${CRYOSPARC_LICENSE_ID}" -o cryosparc_master.tar.gz
+  [ -f "${CRYOSPARC_INSTALL_PATH}/cryosparc_worker.tar.gz" ] || curl -L "https://get.cryosparc.com/download/worker-v4.0.3/${CRYOSPARC_LICENSE_ID}" -o cryosparc_worker.tar.gz
+  
+  # Install cryoSPARC main process
+  tar -xf cryosparc_master.tar.gz
+
+  # cryosparc untars with ownership: 1001:1001 by default. re-align permissions to OSUSER
+  chown -R ${OSUSER}:${OSGROUP} ${CRYOSPARC_INSTALL_PATH}/cryosparc_master
+
+  # Basic configuration for install
+  export CRYOSPARC_FORCE_USER=true
+  export CRYOSPARC_FORCE_HOSTNAME=true
+  export CRYOSPARC_DISABLE_IMPORT_ON_MASTER=true
+
+  # Install Main
+  /bin/su -c "cd ${CRYOSPARC_INSTALL_PATH}/cryosparc_master && ./install.sh --license "${CRYOSPARC_LICENSE_ID}" \
+      --hostname "${HOSTNAME}" \
+      --dbpath "${CRYOSPARC_INSTALL_PATH}"/cryosparc_db \
+      --port 45000 \
+      --allowroot \
+      --yes" - $OSUSER
+  
+  # Enforce configuration long-term
+  echo "export CRYOSPARC_FORCE_USER=true" >> "${CRYOSPARC_INSTALL_PATH}"/cryosparc_master/config.sh
+  echo "export CRYOSPARC_FORCE_HOSTNAME=true" >> "${CRYOSPARC_INSTALL_PATH}"/cryosparc_master/config.sh
+  echo "export CRYOSPARC_DISABLE_IMPORT_ON_MASTER=true" >> "${CRYOSPARC_INSTALL_PATH}"/cryosparc_master/config.sh
+
+  # Ownership of this path determines how cryosparc is started
+  chown -R ${OSUSER}:${OSGROUP} ${CRYOSPARC_INSTALL_PATH}/cryosparc_master
+
+  # Start cryoSPARC main package 
+  /bin/su -c "${CRYOSPARC_INSTALL_PATH}/cryosparc_master/bin/cryosparcm start" - ${OSUSER}
+
+  # Install cryoSPARC worker package
+  cd "${CRYOSPARC_INSTALL_PATH}" || return
+  tar -xf cryosparc_worker.tar.gz
+  chown -R ${OSUSER}:${OSGROUP} ${CRYOSPARC_INSTALL_PATH}/cryosparc_worker
+  /bin/su -c "cd ${CRYOSPARC_INSTALL_PATH}/cryosparc_worker && ./install.sh --license "${CRYOSPARC_LICENSE_ID}" \
+      --cudapath "${CUDA_INSTALL_PATH}/${CUDA_VERSION}" \
+      --yes" - $OSUSER
+  
+  #rm "${CRYOSPARC_INSTALL_PATH}"/*.tar.gz
```

Comment:
Why wouldn't we clean up the .gz?

Reply:
This new version of the script builds CryoSPARC as a systemd service to self-update to latest releases. The .tar.gz is used later in the build script (L347-356):

```
# Update to latest version of CryoSPARC
systemctl stop cryosparc-supervisor.service
/bin/su -c "cd ${CRYOSPARC_INSTALL_PATH} && ${CRYOSPARC_INSTALL_PATH}/cryosparc_master/bin/cryosparcm update" - ${OSUSER}
# Depends on cryosparcm update to pull latest worker to cryosparc_master dir. 
/bin/su -c "cd ${CRYOSPARC_INSTALL_PATH} && cp ${CRYOSPARC_INSTALL_PATH}/cryosparc_master/cryosparc_worker.tar.gz ${CRYOSPARC_INSTALL_PATH}/cryosparc_worker/cryosparc_worker.tar.gz" - ${OSUSER}
# Only update workers if they were installed (continue otherwise)
/bin/su -c "cd ${CRYOSPARC_INSTALL_PATH} && ${CRYOSPARC_INSTALL_PATH}/cryosparc_worker/bin/cryosparcw update" - ${OSUSER} || true 
/bin/su -c "cd ${CRYOSPARC_INSTALL_PATH} && ${CRYOSPARC_INSTALL_PATH}/cryosparc_master/bin/cryosparcm stop" - ${OSUSER}
systemctl start cryosparc-supervisor.service
set -e 
```

Reply:
Let's move the cleanup to after its last use then rather than just commenting it out here :)

Reply:
Done.